### PR TITLE
14.form

### PR DESCRIPTION
**Pull request by cardel**
image

### CODE REVIEW

File: cypress/page/form-fill.page.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    this.lastName = personalInformation.lastName;
+    this.email = personalInformation.email;
+    this.gender = personalInformation.gender;
+    this.mobileNumber = personalInformation.mobileNumber;
+    this.gender = personalInformation.gender;
+    this.dateOfBirth = personalInformation.dateOfBirth;
+    this.hobbies = personalInformation.hobbies;
+    this.currentAddress = personalInformation.currentAddress;
+    this.state = personalInformation.state;
+    this.city = personalInformation.city;
```
Comment (holgiosalos) [inline cypress/page/form-fill.page.ts]: here we should initialize the selectors/locators for each field

File: cypress/page/form-fill.page.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  }  
+
+  public fillForm(): void {
```
Comment (holgiosalos) [inline cypress/page/form-fill.page.ts]: here we should receive the personalInformation object and use that info to fill the form (with the type, click or find commands), withing each cy.get command we should use the selectors/locators, not the data of the personalInformation object

File: cypress/page/form-fill.page.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    cy.get(this.state).type(`${this.state} {enter}`,{force: true});
+    cy.get(this.city).type(`${this.city} {enter}`, {force: true});
+  }
+
+  private getPage() : void {
+    cy.visit("https://demoqa.com/automation-practice-form");
+  }
+
+  public verifySubmission(msg: string): void {
+    cy.get("#example-modal-sizes-title-lg").should("have.text", msg);
```
Comment (holgiosalos) [inline cypress/page/form-fill.page.ts]: `"#example-modal-sizes-title-lg"` should be stored in a class property as it is a selector

File: cypress/page/form-fill.page.ts (hunk 1 above)
Comment (cardel) [reply cypress/page/form-fill.page.ts]: Fixed

File: cypress/page/form-fill.page.ts (hunk 3 above)
Comment (cardel) [reply cypress/page/form-fill.page.ts]: Fixed

File: cypress/page/form-fill.page.ts (hunk 2 above)
Comment (cardel) [reply cypress/page/form-fill.page.ts]: Solved

**Review by holgiosalos — APPROVED** [review]

**Pull request merged**